src/csharp/Microsoft.Spark/Utils/CommandSerDe.cs

```diff
@@ -212,7 +212,7 @@ private sealed class UdfWrapperData
 
             foreach (UdfSerDe.FieldData field in fields)
             {
-                SerializeUdfs((Delegate)field.Value, curNode, udfWrapperNodes, udfs);
+                SerializeUdfs((Delegate)field.ValueData?.Value, curNode, udfWrapperNodes, udfs);
```

Comment:
I think `ValueData` is always non-null. Should we either assert or  `field.ValueData.Value`? Otherwise, we will get null exception down the stream, so it may be a bit confusing.

Reply:
ValueData can potentially be null.  I updated the UdfSerDeTests with an example.

Reply:
Which test is covering this? Can `ValueData` be null in this context? Note that it's a delegate in this context (not capture). Can you check comments in line 70 in this file?

Reply:
Oh you're right.  My tests do not touch `CommandSerde.cs` .  Looks like it only gets to this case when the current node is a non-UDF, a MapUdfWrapper or WorkerChainHelper.  I'll revert this change.